app/assets/javascripts/oxalis/model/sagas/save_saga.js

```diff
@@ -175,6 +175,20 @@ export function compactUpdateActions(updateActionsBatches: Array<Array<UpdateAct
   });
 
 
+  // This part of the code detects deleted trees.
+  // Instead of sending deleteNode/deleteEdge update actions for all nodes of a deleted tree,
+  // just one deleteTree update action is sufficient for the server to delete the tree.
+  // As the deleteTree update action is already part of the update actions if a tree is deleted,
+  // all corresponding deleteNode/deleteEdge update actions can simply be removed.
+  result = result.map((batch) => {
```

Comment:
Personal opinion here, but I'd prefer to have result still as `const` and just append another `map` call to the definition. I think `let` variables always add a mental burden since you have to be aware of where the variable binding is changed.
Maybe it also makes sense to extract the functions passed to map into separate named functions. So that it would read like
`const result = updateActionsBatches.map(compactMovedNodesAndEdges).map(compactDeletedTrees)`

I don't know if the functions need other local variables, though.